spaceship/lib/spaceship/connect_api/client.rb

```diff
@@ -0,0 +1,190 @@
+      def get(url_or_path, params = nil)
+        response = request(:get) do |req|
+          req.url(url_or_path)
+          req.options.params_encoder = Faraday::NestedParamsEncoder
+          req.params = params if params
+          req.headers['Content-Type'] = 'application/json'
+        end
+        handle_response(response)
+      end
+
+      def post(url_or_path, body)
+        response = request(:post) do |req|
+          req.url(url_or_path)
+          req.body = body.to_json
+          req.headers['Content-Type'] = 'application/json'
+        end
+        handle_response(response)
+      end
+
+      def patch(url_or_path, body)
+        response = request(:patch) do |req|
+          req.url(url_or_path)
+          req.body = body.to_json
+          req.headers['Content-Type'] = 'application/json'
+        end
+        handle_response(response)
+      end
+
+      def delete(url_or_path, params = nil, body = nil)
+        response = request(:delete) do |req|
+          req.url(url_or_path)
+          req.options.params_encoder = Faraday::NestedParamsEncoder if params
+          req.params = params if params
+          req.body = body.to_json if body
+          req.headers['Content-Type'] = 'application/json' if body
+        end
+        handle_response(response)
+      end
+
+      protected
+
+      def handle_response(response)
+        if (200...300).cover?(response.status) && (response.body.nil? || response.body.empty?)
+          return
+        end
+
+        raise InternalServerError, "Server error got #{response.status}" if (500...600).cover?(response.status)
+
+        unless response.body.kind_of?(Hash)
+          raise UnexpectedResponse, response.body
+        end
+
+        raise UnexpectedResponse, response.body['error'] if response.body['error']
+
+        raise UnexpectedResponse, handle_errors(response) if response.body['errors']
+
+        raise UnexpectedResponse, "Temporary App Store Connect error: #{response.body}" if response.body['statusCode'] == 'ERROR'
+
+        return Spaceship::ConnectAPI::Response.new(body: response.body, status: response.status, client: self)
+      end
+
+      def handle_errors(response)
+        # Example error format
+        # {
+        #   "errors" : [ {
+        #     "id" : "ce8c391e-f858-411b-a14b-5aa26e0915f2",
+        #     "status" : "400",
+        #     "code" : "PARAMETER_ERROR.INVALID",
+        #     "title" : "A parameter has an invalid value",
+        #     "detail" : "'uploadedDate3' is not a valid field name",
+        #     "source" : {
+        #       "parameter" : "sort"
+        #     }
+        #   } ]
+        # }
+
+        return response.body['errors'].map do |error|
+          "#{error['title']} - #{error['detail']}"
+        end.join(" ")
+      end
+
+      private
+
+      # used to assert all of the named parameters are supplied values
+      #
+      # @raises NameError if the values are nil
+      def assert_required_params(method_name, binding)
```

Comment:
Looks like these aren't used (and possibly defined in the superclass?) I think we should remove them from this implementation.

Reply:
Removed! 💣ed